src/Symfony/Component/AutoMapper/Loader/FileLoader.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AutoMapper\Loader;
+
+use PhpParser\PrettyPrinter\Standard;
+use Symfony\Component\AutoMapper\Generator\Generator;
+use Symfony\Component\AutoMapper\MapperGeneratorMetadataInterface;
+
+class FileLoader implements ClassLoaderInterface
+{
+    private $generator;
+
+    private $directory;
+
+    private $hotReload;
+
+    private $printer;
+
+    private $registry;
+
+    public function __construct(Generator $generator, string $directory, bool $hotReload = true)
+    {
+        $this->generator = $generator;
+        $this->directory = $directory;
+        $this->hotReload = $hotReload;
+        $this->printer = new Standard();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadClass(MapperGeneratorMetadataInterface $mapperGeneratorMetadata): void
+    {
+        $className = $mapperGeneratorMetadata->getMapperClassName();
+        $classPath = $this->directory.\DIRECTORY_SEPARATOR.$className.'.php';
+
+        if (!$this->hotReload) {
+            require $classPath;
+        }
+
+        $hash = $mapperGeneratorMetadata->getHash();
+        $registry = $this->getRegistry();
+
+        if (!isset($registry[$className]) || $registry[$className] !== $hash || !file_exists($classPath)) {
```

Comment:
Couldn't we use the existing Symfony file tracking infrastructure instead?

Reply:
I'm not familiar with that system, can you give me some hint where i can find other code using this ?

Reply:
ping @dunglas